### DISCUSSION TITLE
Is it possible to use references for many to many relationships?

### DISCUSSION THREAD

**Discussion by Shelamkoff** [cycle/orm#338] [Q&A] [answered]
Hi, an exception is thrown when trying to save a relationship using a reference object.
when attempting to store a relationship using a reference object, the following error occurs: "Typed property Cycle\ORM\Transaction\Tuple::$node must not be accessed before initialization."
```php
foreach (json_decode($data['tags'], true) as $tag) {
    $post->tags->add(isset($tag['id']) ?
        new Reference('tag', ['id' => $tag['id']])
        : new Tag($tag['name'])
   );
}
$entityManager->persist($post, true)->run();
````

**Comment by roxblnfk** [comment, marked as answer]
I'm not sure that should work in the current implementation.
It looks like good feature request: supporting for case when subitem of HasMany/ManyToMany relation can be Reference

**Reply by longshadowdev** [reply]
Hello @roxblnfk does that mean that the only way to create a new record that has a ManyToMany to existing records is to load those existing records and assign them to the ManyToMany property? This would have a pretty severe scalability issue in some cases.

Can you think of a workaround, like exposing a OneToMany relationship to the join table and using references to it when saving/updating a record?

Our API allows a user to give an array of primary keys for many to many relationships, we then would ideally create reference objects for each of these IDs instead of loading them into memory just to create the relationship.

Thanks

**Reply by roxblnfk** [reply]
I'll check this possibility
https://github.com/cycle/orm/issues/363

**Reply by JeremyM1985** [reply]
@roxblnfk  was there any work around for this? I have the same issue.

In my case i map the orm to a domain entity and only pass the ids of the many-to-many items.

Many-to-many associations are added/removed through the domain entity. I need to keep the business logic and persistence logic separate and assumed i could use a reference when syncing the many-to-many items.

**Reply by JeremyM1985** [reply]
@longshadowdev did you find a solution to this?

**Reply by roxblnfk** [reply]
@JeremyM1985 sorry, I haven't researched this issue yet.

**Reply by longshadowdev** [reply]
> @longshadowdev did you find a solution to this?

Sorry I haven't used cycle in a few years and can't remember if I found a workaround. I may have tried to load the entities with only their IDs hydrated or something like that.
